### PR TITLE
TINY-10574: now `highlight_on_focus` default is `true`

### PR DESCRIPTION
**Pull request by lorenzo-pomili**
Related Ticket: TINY-10574

Description of Changes:
switch `highlight_on_focus` default to `true`

Pre-checks:
* [x] Changelog entry added
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (DOC-2245)

GitHub issues (if applicable):

### CODE REVIEW

File: .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Changed
+body: now `highlight_on_focus` default is true
```
Comment (TheSpyder) [inline .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml:3]: This is a relatively new option, I think we still need to explain how this change will impact integrators
```suggestion
body: The `highlight_on_focus` option now defaults to true, adding a focus outline to every editor
```

**Review by TheSpyder — APPROVED** [review]

File: .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml (hunk 1 above)
Comment (lorenzo-pomili) [reply .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml]: done

File: modules/tinymce/src/core/test/ts/browser/focus/HighlightOnFocusTest.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ describe('browser.tinymce.core.focus.HighlightOnFocus', () => {
     assertIsHighlighted(editor);
   };
 
-  context('with highlight_on_focus: true', () => {
+  context('without highlight_on_focus option', () => {
     const hook = TinyHooks.bddSetup<Editor>({
       base_url: '/project/tinymce/js/tinymce',
       toolbar: 'forecolor',
```
Comment (HAFRMO) [inline modules/tinymce/src/core/test/ts/browser/focus/HighlightOnFocusTest.ts:30]: Should you remove the ` highlight_on_focus: true` just beneath?

**Review by HAFRMO — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/focus/HighlightOnFocusTest.ts (hunk 2 above)
Comment (lorenzo-pomili) [reply modules/tinymce/src/core/test/ts/browser/focus/HighlightOnFocusTest.ts]: yes, done

File: .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Changed
+body: The `highlight_on_focus` option now defaults to true, adding a focus outline to every editor
```
Comment (spocke) [inline .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml:3]: I guess we should still have a `.` at the end here. We used to have a rule about adding that.

**Review by spocke — APPROVED** [review]

File: .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml (hunk 3 above)
Comment (lorenzo-pomili) [reply .changes/unreleased/tinymce-TINY-10574-2024-01-15.yaml]: added

**Pull request merged**